src/backend/metal/src/device.rs

```diff
@@ -113,6 +113,60 @@ fn get_final_function(library: &metal::LibraryRef, entry: &str, specialization:
     Ok(mtl_function)
 }
 
+pub fn create_depth_stencil_state(
+    device: &metal::DeviceRef,
+    desc: &pso::DepthStencilDesc,
+) -> native::DepthStencilState {
+    let raw = metal::DepthStencilDescriptor::new();
+
+    match desc.depth {
+        pso::DepthTest::On { fun, write } => {
```

Comment:
makes me think that maybe the On/Off types don't work that well, and having an `Option<SomeStruct>` would end up being simpler and more convenient

Reply:
It's not too bad IMO, we could improve it with a helper function for these kinds of cases